src/main/java/dev/waterdog/waterdogpe/ProxyServer.java

```diff
@@ -190,9 +191,22 @@ private void boot() {
         }
 
         this.bedrockServer = new BedrockServer(bindAddress, Runtime.getRuntime().availableProcessors(), this.bossEventLoopGroup, this.workerEventLoopGroup, false);
-        this.bedrockServer.setHandler(new ProxyListener(this));
+        this.bedrockServer.setHandler(new ProxyListener(this, this.queryHandler, bindAddress));
         this.bedrockServer.bind().join();
 
+        for (Integer port : this.getConfiguration().getAdditionalPorts()) {
+            logger.info("Starting additional port " + port);
+            InetSocketAddress additionalBind = new InetSocketAddress(bindAddress.getAddress(), port);
+
+            BedrockServer newServer = new BedrockServer(additionalBind, Runtime.getRuntime().availableProcessors(), this.bossEventLoopGroup, this.workerEventLoopGroup, false);
```

Comment:
I'm still wondering whether its a good idea to assign availableProcessors here, I might actually consider adding configurations for the amount of resources you want to allocate to each individual BedrockServer instance.